cps-wallet-connector/README.md

```diff
@@ -0,0 +1,221 @@
+With the connection standard defining how the wallet and dApp initiate communication, this could be over injected Javascript object, WebRTC, WebTorrent, etc.
+The API defines what communications look like between the dApp and wallet post connection.
+
+This problem statement is concerned with the issues surrounding Cardano's current and future wallet-web bridges.
+These interfaces are difficult to define and historically have been even harder to iterate upon.
+We wish to provide a comprehensive catalog of the current offerings; and their drawbacks to be able to make suggestions on future standards.
+
+### Acknowledgments
+
+<details>
+  <summary><strong>First workshop - 2023-11-27</strong></summary>
+
+  I would like to thank those that contributed to the first workshop hosted by Adam Dean and Ryan Williams ([see presentation slides with notes](https://docs.google.com/presentation/d/1lXw1yMf-Hyp1eVfaFnbTV9897r6pZEd6zpvLNMqT3cI/)).
+  - Beatrice Anihiri
+  - Denis Kalinin
+  - Evgenii Lisitskii
+  - George APEX Pool
+  - George Flerovsky
+  - George Humphreys
+  - Hernán Rajchert
+  - Jack Rousian
+  - Joshua Marchand
+  - Ken Fritschy
+  - Ken-Erik Ølmheim
+  - Leo H
+  - Marcel Baumberg
+  - Martynas Kazlaukas
+  - Michael Chappell
+  - Mircea Hasegan
+  - Nicolas Ayotte
+  - Rhys Bartels-Waller
+  - Robert Phair
+  - Rodolpho Ribeiro
+  - Steven Johnson
+  - Teodora Sevastru Lunn
+  - Thomas Lindseth
+  - Thomas Upfield
+  - Vladimir Kalnitsky
+
+</details>
+
+
+## Problem
+<!-- A more elaborate description of the problem and its context. This section should explain what motivates the writing of the CPS document. -->
+
+The core motivation of this document is to catalog the current state of Cardano's wallet connectors, discuss their flaws and identify keys concerns for future connector authors to be aware of.
+We hope that by discussing the issues we will inspire the next generation of Cardano wallet connectors so that the ecosystem can grow beyond it's first iterations.
+
+Ineffective dApp-connectors can cause a range of issues for wallets, dApp developers and thus users.
+Due to the nature of these connections there can often be unforeseen impacts of small design decisions.
+
+For users, connectors should be widely supported across platforms, secure and decentralized.
+For dApps, connectors should be expressive and extendable.
+For wallets, connectors should be secure, extendable and should not expect wallet's to go beyond their standard activities.
+
+### Core Concerns
+
+#### 1. Secure connection
+Security should remain of paramount concern within web 3.
+Connection standards must strive to above all not compromise the security of wallets or dApps.
+Connection standards should be aware of potential impacts of standard security vulnerabilities, such as man-in-middle attacks.
+Furthermore, standards should seek to be censorship resistant, by an ecosystem offering multiple connection standards we move in this direction.
```

Comment:
As a citizen of a country with somewhat visible censorship, I still fail to see how a wallet standard can do anything about it, especially by an ecosystem offering many competing standards :slightly_smiling_face:

Reply:
@klntsky @Ryun1 I would stand by this statement as a citizen of two countries with "invisible" but constant censorship, where the governments brag about their incipient capabilities to control unorthodox behaviour through financial restrictions.  In a similarly composed third country (Canada) such policies have already been executed and we will keep seeing more of it whenever exposed personal data touches financial expression (i.e web site visits & dApp use).

So without a rigorous political discussion I think therefore this is well listed as a core security concern.  It may even be an implementation issue since wallet connectors may eventually have to incorporate security countermeasures somehow.

Reply:
@rphair visible or not, what attack vectors are we considering? dApp frontend takeover? wallet takeover? network traffic blocking? I believe only reusable query layers are related, but in this case censorship resistance is achieved by having more query layer providers rather than many standards.

Reply:
@Ryun1 I'd rather see more information about intrusion by well-funded adversaries, to qualify the importance of "censorship resistance", rather than taking that statement out as being underqualified or ambiguous.  The items mentioned by @klntsky would be a great place to start a brief illustration of the matter, I think.

Reply:
sure, I think the latter statement could be added as a qualification... or the whole point removed, at your discretion...